src/state/slices/opportunitiesSlice/resolvers/cosmosSdk/utils.ts

```diff
@@ -33,3 +41,49 @@ export const makeUniqueValidatorAccountIds = (
       ),
     ]),
   )
+
+export const makeAccountUserData = ({
+  cosmosAccount,
+  validatorIds,
+}: {
+  cosmosAccount: Account<CosmosSdkChainId>
+  validatorIds: ValidatorId[]
+}): OpportunitiesState['userStaking']['byId'] => {
+  const delegations = cosmosAccount.chainSpecific.delegations
+  // const undelegations = cosmosAccount.chainSpecific.delegations
+  const rewards = cosmosAccount.chainSpecific.rewards
+
+  const delegationsByValidator = groupBy(delegations, delegation => delegation.validator.address)
+  // TODO(gomes): handle me in a separate PR, we will need chain-specific userData similar to the savers-specific-metadata
+  // const undelegationsByValidator = groupBy(
+  // undelegations,
+  // undelegation => undelegation.validator.address,
+  // )
+  const rewardsByValidator = groupBy(rewards, reward => reward.validator.address)
+
+  return validatorIds.reduce((acc, validatorId) => {
+    const validatorAddress = fromAccountId(validatorId).account
+    const userStakingId = serializeUserStakingId(
+      toAccountId({ account: cosmosAccount.pubkey, chainId: cosmosAccount.chainId }),
+      validatorId,
+    )
+
+    const maybeValidatorDelegations = bnOrZero(
+      delegationsByValidator[validatorAddress]?.[0]?.amount,
+    )
+    const maybeValidatorRewards = rewardsByValidator[validatorAddress]?.[0]?.rewards
+    const maybeValidatorRewardsAggregated = (maybeValidatorRewards ?? []).reduce(
+      (a, b) => a.plus(b.amount),
+      bn(0),
+    )
+
+    if (maybeValidatorDelegations.gt(0) || maybeValidatorRewardsAggregated.gt(0)) {
```

Comment:
Don't populate `userStaking` for no reason, especially since we're most likely going to use `userStakingId`s as a criteria for active positions